src/modules/payment/ExpiryMessage.tsx

```diff
@@ -0,0 +1,76 @@
+import {MessageInfoText} from '@atb/components/message-info-text';
+import {Statuses, StyleSheet, Theme} from '@atb/theme';
+import {useTranslation} from '@atb/translations';
+import PaymentMethodsTexts from '@atb/translations/screens/subscreens/PaymentMethods';
+import {formatToShortDateWithYear} from '@atb/utils/date';
+import {addDays, parseISO} from 'date-fns';
+import {View} from 'react-native';
+import {RecurringPayment} from '../ticketing';
+
+export const ExpiryMessage = ({
+  recurringPayment,
+}: {
+  recurringPayment?: RecurringPayment;
+}) => {
+  const {t, language} = useTranslation();
+  const styles = useStyles();
+  const now = Date.now();
+  const inThirtyDays = addDays(now, 30).getTime();
+
+  if (!recurringPayment) return null;
+
+  const expiresAt = parseISO(recurringPayment.expiresAt).getTime();
+  const cardExpiresAt = parseISO(recurringPayment.cardExpiresAt).getTime();
+
+  let message: {type: Statuses; message: string} | null = null;
```

Comment:
Maybe make this a `let message = getMessage(recurringPayment);`?

That way you can use `return`s to make the code below cleaner ✨